src/Microsoft.Health.Fhir.SqlServer/Features/Search/Expressions/Visitors/QueryGenerators/CompartmentQueryGeneratorV2.cs

```diff
@@ -0,0 +1,28 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using Microsoft.Health.Fhir.Core.Features.Search.Expressions;
+using Microsoft.Health.Fhir.SqlServer.Features.Schema.Model;
+using Microsoft.Health.SqlServer.Features.Schema.Model;
+
+namespace Microsoft.Health.Fhir.SqlServer.Features.Search.Expressions.Visitors.QueryGenerators
+{
+    internal class CompartmentQueryGeneratorV2 : SearchParamTableExpressionQueryGenerator
+    {
+        public static readonly CompartmentQueryGeneratorV2 Instance = new CompartmentQueryGeneratorV2();
+
+        public override Table Table => VLatest.ReferenceSearchParam;
+
+        public override SearchParameterQueryGeneratorContext VisitCompartment(CompartmentSearchExpression expression, SearchParameterQueryGeneratorContext context)
```

Comment:
Is this still needed?

Reply:
Do you mean the suffix with the V2 and the V1? or the class itself? 
The class [CompartmentQueryGeneratorV2.cs](https://github.com/microsoft/fhir-server/pull/2559/files/916a43b25cef87e4441e148266f31553479327f8#diff-eb905cf3b3e36c8f3d148a7b204e3bf13d14903db186c9eebc076d2ba5c2fb9b) is important because it has the responsibility of telling the ideal table where queries should be on top of.